app/internal/world_clock.py

```diff
@@ -0,0 +1,512 @@
+    """
+    continent_url = f"{TIMEZONES_BASE_URL}/{place_name}"
+    api_data = await get_api_data(continent_url)
+    timezone = api_data[ARB_COUNTRY_TIMEZONE]
+    return f"{TIMEZONES_BASE_URL}/{timezone}"
+
+
+def load_country_subcountry_data_set() -> List[Dict[str, str]]:
+    """Load the country-csubcountry data set.
+
+    Returns:
+        list: The country-subcountry data set.
+    """
+    with open(TIMEZONES_COUNTRY_SUBCOUNTRY_PATH) as file:
+        return json.load(file)
+
+
+def get_timezone_from_subcountry(place_name: str) -> str:
+    """Get the timezone from the subcountry.
+    Args:
+        place_name (str): The given place name.
+    Returns:
+        str: The timezone.
+    """
+    subcountry = get_subcountry(place_name)
+    country = get_country(place_name)
+    if subcountry:
+        for timezone, details in load_country_subcountry_data_set().items():
+            if (details[SUBCOUNTRY_NAME_KEY_IN_CITIES] == subcountry) or (
+                details[COUNTRY] == country
+            ):
+                return f"{TIMEZONES_BASE_URL}/{timezone}"
+
+
+async def get_timezone_path_for_given_place(place_name: str) -> Optional[str]:
+    """Get a timezone path for a given place.
+    Args:
+        place_name (str): The given place name.
+    Returns:
+        str: The timezone path.
+    """
+    possibilities = await get_all_possible_timezone_paths_for_given_place(
+        place_name
+    )
+    timezones = await get_api_data(TIMEZONES_BASE_URL)
+    for possibility in possibilities:
+        if possibility in timezones:
+            return f"{TIMEZONES_BASE_URL}/{possibility}"
+    # No explicit such timezones. Get an arbitrary timezone of country
+    if await standardize_continent(place_name.title()):
+        return await get_arbitrary_timezone_of_country(place_name)
+    return get_timezone_from_subcountry(place_name)
+
+
+async def get_current_time_in_place(place_name: str) -> Optional[str]:
+    """Get the current time in a given place.
+    Args:
+        place_name (str): The given place name.
+    Returns:
+        str: The timezone path.
+    """
+    path = await get_timezone_path_for_given_place(place_name)
+    if not path:
+        return
+    current_datetime_details = await get_api_data(path)
+    current_datetime_full = current_datetime_details[CURRENT_TIME_KEY]
+    current_datetime_parsed = dateutil.parser.parse(current_datetime_full)
+    current_time = current_datetime_parsed.strftime("%H:%M:%S")
+    return current_time
+
+
+def get_part_of_day_and_feedback(time: datetime) -> Tuple[str, str]:
+    """Get the part of day and a suitable feedback for a given time.
+    Args:
+        time (datetime): The given time.
+    Returns:
+        tuple: The part of day description and the feedback.
+    """
+    for part in PARTS_OF_THE_DAY_FEEDBACK:
+        if (
+            time
+            >= datetime.strptime(
+                part.start,
+                "%H:%M:%S",
+            )
```

Comment:
Export this and the other one to a variable, outside the loop, and create 1 line `if` statement